File: rest-service/src/main/java/mb/lib/formatter/model/UserReportRow.java
package mb.lib.formatter.model;

import java.time.OffsetDateTime;

import mb.lib.model.HashID;
import mb.lib.model.JobStatus;
import org.veupathdb.lib.blast.BlastFormatter;

public class UserReportRow extends ReportRow
{
  private final long   userID;
  private final String description;

  public UserReportRow(
    HashID reportID,
    HashID jobID,
    JobStatus status,
    BlastFormatter config,
    int queueID,
    OffsetDateTime createdOn,
    long userID,
    String description
  ) {
    super(reportID, jobID, status, config, queueID, createdOn);
    this.userID      = userID;
    this.description = description;
  }

  public UserReportRow(ReportRow row, long userID, String description) {
    this(
      row.getReportID(),
      row.getJobID(),
      row.getStatus(),
      row.getConfig(),
      row.getQueueID(),
      row.getCreatedOn(),
      userID,
      description
    );
  }

  public long getUserID() {
    return userID;
  }

  public String getDescription() {
    return description;
  }

  @Override
  public String toString() {
    return "UserReportRow{jobID="
      + getJobID()
      + ", reportID="
      + getReportID()
      + ", userID="
      + userID
      + "}";
  }
}

File: rest-service/src/main/kotlin/mb/lib/report/model/UserReportRow.kt
package mb.lib.report.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_DEFAULT
import mb.lib.model.JobStatus
import org.veupathdb.lib.blast.BlastFormatter
import org.veupathdb.lib.hash_id.HashID
import java.time.OffsetDateTime

@JsonInclude(NON_DEFAULT)
class UserReportRow(
  reportID: HashID,
  jobID: HashID,
  status: JobStatus,
  config: BlastFormatter,
  queueID: Int,
  createdOn: OffsetDateTime,
  val userID: Long,
  val description: String?
) : ReportRow(reportID, jobID, status, config, queueID, createdOn) {
  constructor(row: ReportRow, userID: Long, description: String?) : this(
    row.reportID,
    row.jobID,
    row.status,
    row.config,
    row.queueID,
    row.createdOn,
    userID,
    description
  )
}